tests/Doctrine/Tests/ORM/Functional/PaginationTest.php

```diff
@@ -669,6 +669,28 @@ public function testCountQueryStripsParametersInSelect()
         $this->assertCount(9, $paginator);
     }
 
+    /**
+     * @dataProvider useOutputWalkersAndFetchJoinCollection
+     */
+    public function testPaginationWithSubSelectOrderByExpression($useOutputWalker, $fetchJoinCollection)
+    {
+        $query = $this->_em->createQuery(
+            "SELECT u, 
+                (
+                    SELECT MAX(a.version)
+                    FROM Doctrine\\Tests\\Models\\CMS\\CmsArticle a
+                    WHERE a.user = u
+                ) AS HIDDEN max_version
+            FROM Doctrine\\Tests\\Models\\CMS\\CmsUser u
+            ORDER BY max_version DESC"
+        );
+
+        $paginator = new Paginator($query, $fetchJoinCollection);
+        $paginator->setUseOutputWalkers($useOutputWalker);
+
+        $this->assertCount(9, $paginator->getIterator());
```

Comment:
Can you also add an SQL generation test? (see non-functional tests around the paginator)